### ISSUE TITLE
Zone / Schedule Labels

### ISSUE THREAD

**Issue by chasingtheflow** [valiquette/homebridge-rachio-irrigation#65]
Is there a way for this to pull in the existing zone and schedule labels (and ideally images) from rachio? I just set this up naively but everything appears in homekit as "zone 1", "zone 2" etc. even though they have names in rachio. Do you have to maintain the names in each place?

**Comment by valiquette** [comment]
It should be pulling in the names as you have them configured. Make sure they show correct in homebridge first. You may have to remove the accessory in homekit and re add it if it is stuck thinking you want the names to be zone1, zone 2

**Comment by chasingtheflow** [comment]
Yeah, I guess it maybe just took time to sync because they appeared later. Though there are a bunch of unnamed switches still (in addition to my named switches). 

https://github.com/valiquette/homebridge-rachio-irrigation/assets/718368/3e3d4e45-04ff-45b1-805a-ec91502e9e93

**Comment by pponce** [comment]
The schedule names don't get set if you selected to have switches enabled for easy turning on and off of schedules.
The controller standby switch also does not get properly named to something like Hibernate or Standby.
The zone names do get set properly though.
Thanks for this great plugin!

**Comment by valiquette** [comment]
The names for scheduled come from the API response for the controller info. 
In the debug log with the plugin setting to show API responses you should see something like this 
     "scheduleRules": [
       {
       "id": "cc9c6e6f-c285-4a7b-9911-ff6065e7ff5b",
       "name": "something",
       "externalName": "unknown"
       }
Standby and Run All are the device name from the API with "Standby" or "Run All" appended.
Are you just seeing a blank name ?

Could you enable debug logging with API responses shown and attach it ?

**Comment by valiquette** [comment]
When you set up this controller and added the schedules did you set it up from the phone app or the web app?

**Comment by pponce** [comment]
> When you set up this controller and added the schedules did you set it up from the phone app or the web app?
These schedules were likely added years ago via the Rachio phone iOS app.
The configured switches all come in with the correct names that I have them setup with in the Rachio app.
Not sure why they were not set with the names on first setting up the controller with HomeKit.

> Are you just seeing a blank name ?
No they the 3 schedules and the standby switch all had the same generic name. Not remembering what it was now though.
I changed the name in HomeKit already so don't know what the original names were.


Here is my log output with the the API response enabled:
[02/01/2025, 07:57:37] [Rachio] Found Controller CortlandRachio status ONLINE
[02/01/2025, 07:57:37] [Rachio] Getting device state info...
[02/01/2025, 07:57:37] [Rachio] Retrieved device state IDLE for CortlandRachio with a DESIRED_ACTIVE state, running iro2-firmware-5-119
[02/01/2025, 07:57:37] [Rachio] Configuring Rachio webhooks for controller ID XXXXXXXXXXXXXX
[02/01/2025, 07:57:37] [Rachio] Configure Irrigation system for CortlandRachio
[02/01/2025, 07:57:37] [Rachio] Configured zone-1 for Lawn Fence side with 3 min runtime
[02/01/2025, 07:57:37] [Rachio] Configured zone-2 for Lawn Deck side with 3 min runtime
[02/01/2025, 07:57:37] [Rachio] Configured zone-3 for LightWell Bed with 3 min runtime
[02/01/2025, 07:57:37] [Rachio] Configured zone-4 for Main Bed with 3 min runtime
[02/01/2025, 07:57:38] [Rachio] Configured zone-5 for Pots with 3 min runtime
[02/01/2025, 07:57:38] [Rachio] Configured zone-6 for Zone 6 with 3 min runtime
[02/01/2025, 07:57:38] [Rachio] Configured zone-7 for Zone 7 with 3 min runtime
[02/01/2025, 07:57:38] [Rachio] Configured zone-8 for Zone 8 with 3 min runtime
[02/01/2025, 07:57:38] [Rachio] Configured switch for Pots Evening
[02/01/2025, 07:57:38] [Rachio] Configured switch for Lawn Fence side
[02/01/2025, 07:57:38] [Rachio] Configured switch for Beds
[02/01/2025, 07:57:38] [Rachio] Configured switch for CortlandRachio Standby

**Comment by valiquette** [comment]
This does not look like the homebridge debug logging is turned on, but that is OK. The logging from the plugin is showing the names from the API response which are all good. 
Do you see switches with these names on the homebridge accessories page?

**Comment by pponce** [comment]
The switches right now in the home app all have the names that I manually chose for them.
For example the standby switch I named “hibernate” in the home app.

Now that everything is named in the home app there is no issue at all. Just took a minute to turn on the schedules and see which one actually ran and then renamed them.
The issue was only on initial install of the plugin and adding the child bridge to the home app.
Zone names came in fine. But schedules and the standby switch all had generic names. I’m not remembering if they were numbered or all had the same exact names

**Comment by valiquette** [comment]
I think I found something that could be causing this issue.  I published a beta release 1.3.8-beta.0 for you to try?
This was changed while back to not overwrite any name change in HomeKit, I can't test this at this time, but if you set the plugin to not show standby it should remove the switch, then the next time it is added it would have the name "CortlandRachio Standby" you should be able to rename it and your new name will persist any plugin restart.

**Comment by pponce** [comment]
Looks like it works.
-I disabled the standby switch setting in the config.
-restarted the child bridge 
-confirmed the standby switch was removed in the home app.
-Installed the beta version.
-re-enabled the standby switch setting in the config
-switch showed up in the home app as “Cortlandrachio standby”

thanks again for your contributions to the community!

**Comment by valiquette** [comment]
Resolved in 1.4.0
